test/position_calc_test/NLS_guess_distribution.py

```diff
@@ -0,0 +1,211 @@
+
+    @param pinger_guess     Initial guess for pinger position
+    @param noise_stdev      Standard deviation of gaussian noise injected to time differences
+                            (mean is 0)
+    @param n_iters          Number of noise simulations performed before plotting results
+    @param change_var       provides information about which variable is changing in simulation
+                            run to determine what information gets printed in plot title. Takes
+                            type ChangingVariable enum
+    @param plot_error_hist  flag to determine if to output a histogram of angular and distance error
+    '''
+    initial_guess = pinger_guess
+    opt_type = OptimizationType.nelder_mead
+    component = NLS_position_calc(optimization_type=opt_type, initial_guess=initial_guess)
+
+    true_tdoa = tuple(position_calc_utils.tdoa_function_3D(
+                np.array([cfg.pinger_position.r, cfg.pinger_position.phi]), hydrophone_position, True)
+                for hydrophone_position in cfg.hydrophone_positions[1:])
+
+    NLS_outputs = []
+    for i in range(n_iters):
+        noise = [np.random.normal(0, noise_stdev*TIME_CONV['us']) for j in range(len(true_tdoa))]
+        tdoa = [tdoa_val+noise_val for tdoa_val,noise_val in zip(true_tdoa, noise)]
+        predicted_pos = component.apply(tdoa)
+        NLS_outputs.append(predicted_pos)
+
+    r_err = [(pos.r - cfg.pinger_position.r) for pos in NLS_outputs]
+    phi_err = [(pos.phi - cfg.pinger_position.phi)*CONV_2_DEG for pos in NLS_outputs]
+    x = [cyl_2_cart(pos).x for pos in NLS_outputs]
+    y = [cyl_2_cart(pos).y for pos in NLS_outputs]
+
+    if (change_var == ChangingVariable.Noise):
+        change = " Noise " + r'$\sigma = $' + str(round(noise_stdev,1)) + r'$\mu s$'
+    elif (change_var == ChangingVariable.HydrophonePositions):
+        change = ""
+    elif (change_var == ChangingVariable.PingerGuess):
+        change = " Initial Guess: r=" + str(round(pinger_guess.r,1)) + " " + r'$\phi =$' + str(round(pinger_guess.phi*CONV_2_DEG,1))
+    else:
+        change = " Pinger Position: r=" + str(round(cfg.pinger_position.r,1)) + " " + r'$\phi =$' + str(round(cfg.pinger_position.phi*CONV_2_DEG,1)) + " " + "z=" + str(round(cfg.pinger_position.z,1))
+
+    if (plot_error_hist):
+        f, (ax1, ax2) = plt.subplots(1, 2, sharey=True)
+        f.suptitle("NLS Error Distribution" + change)
+        ax1.hist(r_err, histtype='step', density=True)
+        ax1.set_title("Radial Distance Error")
+        ax1.set_xlabel(r'$\Delta r$ (m)')
+        ax1.set_ylabel("Normalized Count")
+        ax2.hist(phi_err, histtype='step', density=True)    
+        ax2.set_title("Angular Error")
+        ax2.set_xlabel(r'$\Delta \phi (^\circ)$')
+
+    hx = [cyl_2_cart(pos).x for pos in cfg.hydrophone_positions]
+    hy = [cyl_2_cart(pos).y for pos in cfg.hydrophone_positions]
+
+    f, (ax1, ax2) = plt.subplots(1, 2, figsize=(10,5))
```

Comment:
I personally always preferred having a separate function for all the plotting stuff, if for no other reason than it's just nicer. Not a big deal